content/developer/tutorials/server_framework_101/05_connect_the_dots.rst

```diff
@@ -0,0 +1,200 @@
+----------------------
+
+.. todo: change section title
+
+tmp
+
+.. todo: implement search method
+.. todo: api.depends triggers a recomputation when update (already said above)
+
+.. _tutorials/server_framework_101/onchanges:
+
+Onchange methods
+================
+
+.. todo: change section title
+
+**Onchange methods** are a feature of the server framework designed to respond to changes in field
+values directly within the user interface. They execute when a user modifies a field in a form view,
+even before saving the record to the database. This allows for real-time updates of other fields and
+provides immediate user feedback, such as validations, warnings, or suggestions. However, because
+onchange methods are only triggered by changes made in the UI, specifically from a form view, they
+are best suited for assisting with data entry and providing feedback, rather than implementing core
+business logic in a module.
+
+In Odoo, onchange methods are implemented as Python methods and linked to one or more fields using
+the :code:`@api.onchange()` decorator. These methods are triggered the specified fields' values are
+altered and operate on the in-memory representation of a single-record recordset received through
+`self`. If field values are modified, the changes are automatically reflected in the UI.
+Additionally, onchange methods can return non-blocking warnings or raise blocking errors to guide
+users during data entry.
+
+
+.. todo: raise UserError + translation
+.. todo: if garden checked -> show and compute total area
+
+tmp
+
+.. _tutorials/server_framework_101/constraints:
+
+Constraints
+===========
+
+.. todo: change section title
+
+tmp
+
+.. _tutorials/server_framework_101/sql_constraints:
+
+SQL constraints
+---------------
+
+tmp
+
+.. todo: price more than zero
+.. todo: unique tag constraint
+
+.. _tutorials/server_framework_101/python_constraints:
+
+Python constraints
+------------------
+
+tmp
+
+.. todo: accept only one offer
+
+.. _tutorials/server_framework_101/defaults:
+
+Defaults
+========
+
+.. todo: change section title
+.. todo: introduce lambda functions for defaults :point_down:
+   also mention that `self` is evaluated as the current recordset in lambda functions
+
+There is a problem with the way we defined our `Date` fields in the previous chapters: their default value relies on
+:code:`fields.Date.today()` or some other static method. When the code is loaded into memory, the date is
+computed once and reused for all newly created records until the server is shut down. You probably didn't
+notice it, unless you kept your server running for several days, but it would be much more visible with
+`Datetime` fields, as all newly created records would share the same timestamp.
+
+That's where lambda functions come in handy. As they generate an anonymous function each time they're evaluated
+at runtime, they can be used in the computation of default field values to return an updated value for each new record.
```

Comment:
you do not want to mention that there is still built-in  `fields.Date.today` that is equivalent of lambda?

Reply:
Yes, it's planned, indeed. I need to find a better approach to introduce lambda functions without making it over-engineered.